File: day01/main.py
import numpy as np

print('Part 1:', sum(np.diff(np.loadtxt('input.txt')) > 0))

arr = np.convolve(np.loadtxt('input.txt'), np.ones(3, dtype=np.int), mode='valid')
print('Part 2:', sum(np.diff(arr) > 0))

File: day01/R/main.R
val <- read.csv('input.txt', header = FALSE)$V1
cat('Part 1:', sum(diff(val) > 0))
n <- length(val)
cat('Part 2:', sum(diff(val[1:(n-2)] + val[2:(n-1)] + val[3:n]) > 0))
